core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -964,7 +964,7 @@ public TerminationHandle warmTopReader(final IndexShard indexShard, IndexMetaDat
             final Map<String, MappedFieldType> warmUpGlobalOrdinals = new HashMap<>();
             for (DocumentMapper docMapper : mapperService.docMappers(false)) {
                 for (FieldMapper fieldMapper : docMapper.mappers()) {
-                    final FieldDataType fieldDataType = fieldMapper.fieldType().fieldDataType();
+                    final FieldDataType fieldDataType = fieldMapper.fieldDataType();
```

Comment:
I'm not a big fan of this exception logic. This works out now because `_parent` field name is mapped to a special field data implementation in the field data service.